qiskit_experiments/library/characterization/zz_ramsey.py

```diff
@@ -0,0 +1,339 @@
+            dt: the conversion factor from the units used in the circuits'
+                delay instructions to time in seconds.
+            delay: the length of each delay instruction in each circuit
+
+        The circuits are parameterized with these parameters so that
+        ``QuantumCircuit.draw()`` produces a nice representation of the
+        parameters in the circuit.
+
+        Args:
+            delay_unit: the unit of circuit delay instructions.
+
+        Returns:
+            Circuits for series 0 and 1
+        """
+        metadata = {
+            "experiment_type": self._type,
+            "qubits": self.physical_qubits,
+            "unit": "s",
+        }
+
+        delay = Parameter("delay")
+        freq = Parameter("f")
+        dt = Parameter("dt")
+        pi = Parameter("pi")
+
+        delay_time_total = 2 * delay * dt
+
+        # Template circuit for series 0
+        # Control qubit starting in |0> state, flipping to |1> in middle
+        circ0 = QuantumCircuit(2, 1, metadata=metadata.copy())
+        circ0.metadata["series"] = "0"
+
+        circ0.sx(0)
+
+        circ0.barrier()
+        circ0.delay(delay, 0, delay_unit)
+        circ0.barrier()
+
+        circ0.x(0)
+        circ0.x(1)
+
+        circ0.barrier()
+        circ0.delay(delay, 0, delay_unit)
+        circ0.barrier()
+
+        circ0.rz(2 * pi * freq * delay_time_total, 0)
+        circ0.sx(0)
+        # Flip control back to 0, so control qubit is in 0 for all circuits
+        # when qubit 1 is measured.
+        circ0.x(1)
+
+        circ0.barrier()
+        circ0.measure(0, 0)
+
+        # Template circuit for series 1
+        # Control qubit starting in |1> state, flipping to |0> in middle
+        circ1 = QuantumCircuit(2, 1, metadata=metadata.copy())
+        circ1.metadata["series"] = "1"
+
+        circ1.x(1)
+        circ1.sx(0)
+
+        circ1.barrier()
+        circ1.delay(delay, 0, delay_unit)
+        circ1.barrier()
+
+        circ1.x(0)
+        circ1.x(1)
+
+        circ1.barrier()
+        circ1.delay(delay, 0, delay_unit)
+        circ1.barrier()
+
+        circ1.rz(2 * pi * freq * delay_time_total, 0)
+        circ1.sx(0)
+
+        circ1.barrier()
+        circ1.measure(0, 0)
+
+        return circ0, circ1
+
+    def _template_circuits(
+        self, dt_value: float = 1.0, delay_unit: str = "s"
+    ) -> Tuple[QuantumCircuit, QuantumCircuit]:
+        """Template circuits for series 0 and 1 parameterized by delay
+
+        The generated circuits have the length of the delay instructions as the
+        only parameter.
+
+        Args:
+            dt_value: the value of the backend ``dt`` value. Used to convert
+                delay values into units of seconds.
+            delay_unit: the unit of circuit delay instructions.
+
+        Returns:
+            Circuits for series 0 and 1
+        """
+        circ0, circ1 = self._parameterized_circuits(delay_unit=delay_unit)
```

Comment:
Why we cannot directly generate these circuits inside the method? Personally I don't like having too many (unnecessary) helper functions because it makes reading the code harder. Do you plan to add another experiment with variant of ZZRamsey? Parameterizing `pi` and `dt` for handling of missing dt seems overengineering. Alternatively you can call timing helper also inside `_parameterized_circuits` becasue this method knows `self.backend`.

Reply:
The main purpose of splitting `_parameterized_circuits` from `_template_circuits` is that I want a method that produces circuits that print nicely. `_parameterized_circuits` was used to generate the circuit representation in the doc string for example. I tried to make this more clear in a2232308be773d47b2d140e030678f27e94f7fe7 by shifting the parameterizing out into its own method, so it is not part of the `.circuits()` chain.